From d58a763f8f99e0daefaf6da7e96d2e3858fdcdf4 Mon Sep 17 00:00:00 2001
From: Eric Covener <covener@apache.org>
Date: Fri, 20 Mar 2020 13:00:44 +0000
Subject: [PATCH] Merge r1875429 from trunk:

fix apxs for non-destdir builds

subtr(...,0, -1) chops off a character and leaves relative path.

(you will need to re-run configure)



Reviewed By: +1: covener, gbechis, rpluem



git-svn-id: https://svn.apache.org/repos/asf/httpd/httpd/branches/2.4.x@1875475 13f79535-47bb-0310-9956-ffa450edef68
---
 support/apxs.in | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/support/apxs.in b/support/apxs.in
index f7110978e6a..b2705faf5b7 100644
--- a/support/apxs.in
+++ b/support/apxs.in
@@ -26,7 +26,12 @@ package apxs;
 #   are we building in a cross compile environment? If so, destdir contains
 #   the base directory of the cross compiled environment, otherwise destdir
 #   is the empty string.
-my $destdir = substr($0, 0, rindex($0, "@exp_bindir@"));
+
+my $destdir = "";
+my $ddi = rindex($0, "@exp_bindir@");
+if ($ddi >= 0) {
+  $destdir = substr($0, 0, $ddi);
+}
 
 my %config_vars = ();
 
